src/components/document/document-dnd-context.tsx

```diff
@@ -0,0 +1,27 @@
+import React from "react";
+import {
+  DndContext, DragEndEvent, DragStartEvent, PointerSensor, useSensor, useSensors
+} from "@dnd-kit/core";
+
+import { useUIStore } from "../../hooks/use-stores";
+
+interface IDocumentDndContextProps {
+  children: any;
+}
+export const DocumentDndContext = ({ children }: IDocumentDndContextProps) => {
+  const ui = useUIStore();
+  const sensors = useSensors(
+    useSensor(PointerSensor, {
+      activationConstraint: {
+        distance: 15
+      }
+    })
+  );
+  const onDragStart = (event: DragStartEvent) => ui.setDraggingId(event.active.id.toString());
+  const onDragEnd = (event: DragEndEvent) => ui.setDraggingId(undefined);
```

Comment:
I like this -- we may have to adopt this in CODAP. 😉

Reply:
I'll be honest, I mostly broke this out into a separate component because `DocumentContentComponent` is a class component, and I needed to use `useSensors`. But I agree it's a good change that cleans the code up nicely.